### ISSUE TITLE
How to specify the secret version for Cloud Build webook

### ISSUE THREAD

**Issue by hpe-ii-kaito** [GoogleCloudPlatform/k8s-config-connector#843]
### Describe your question

Hi team,

I'm currently working on Cloud Build Trigger with webhook and struggling to specify the secret version defined in SecretManagerSecretVersion. Any ideas how I can specify the secret? 

```
Events:
  Type     Reason              Age                From                          Message
  ----     ------              ----               ----                          -------
  Warning  DependencyNotFound  12m                cloudbuildtrigger-controller  reference SecretManagerSecret torishin-dev01-control/webhook-secret is not found
  Normal   Updating            5s (x12 over 12m)  cloudbuildtrigger-controller  Update in progress
  Warning  UpdateFailed        5s (x12 over 12m)  cloudbuildtrigger-controller  Update call failed: error applying desired state: summary: Error creating Trigger: googleapi: Error 400: service account service-xxxxx@gcp-sa-cloudbuild.iam.gserviceaccount.com failed to access secret version for webhook trigger: generic::invalid_argument: The provided Secret Version ID [projects/xxxxxx/secrets/webhook-secret] does not match the expected format [projects/*/secrets/*/versions/*]
```

Here is the minimal config that I used.

```
apiVersion: cloudbuild.cnrm.cloud.google.com/v1beta1
kind: CloudBuildTrigger
metadata:
  name: test-trigger
spec:
  disabled: false
  build:
    step:
      - name: gcr.io/cloud-builders/git
        script: >
          #!/usr/bin/env bash
          echo hello
  webhookConfig:
    secretRef:
      name: webhook-secret
---
apiVersion: secretmanager.cnrm.cloud.google.com/v1beta1
kind: SecretManagerSecret
metadata:
  name: webhook-secret
  labels:
    replication-type: automatic
spec:
  replication:
    automatic: true
---
apiVersion: secretmanager.cnrm.cloud.google.com/v1beta1
kind: SecretManagerSecretVersion
metadata:
  name: webhook-secret
spec:
  enabled: true
  secretData:
    value: webhook
  secretRef:
    name: webhook-secret
```

**Comment by hpe-ii-kaito** [comment]
As I was looking at the terraform documentation, I noticed that the webhookConfig secret is referencing secret_manager_secret_version.  But in the case of config connector, it is referring to SecretManagerSecret. Is this intentional or a bug?

https://registry.terraform.io/providers/hashicorp/google/latest/docs/resources/cloudbuild_trigger#example-usage---cloudbuild-trigger-webhook-config

**Comment by hpe-ii-kaito** [comment]
> Were you able to use secretRef.external and specify a URL of SecretManagerSecretVersion? Did that work?
Yes, it works when I use use secretRef.external with SecretManagerSecretVersion URL. So I'm suspecting that  secretRef.name is misconfigured.

**Comment by diviner524** [comment]
This is fixed and will be released on our next version. (1.109.0)

**Comment by diviner524** [comment]
Fixed in 1.109.0
